org.eclipse.january/src/org/eclipse/january/dataset/DatasetUtils.java

```diff
@@ -476,7 +476,7 @@ public static Dataset concatenate(final IDataset[] as, final int axis) {
 	public static List<Dataset> split(final Dataset a, int[] indices, final int axis) {
 		final int[] ashape = a.getShapeRef();
 		final int rank = ashape.length;
-		if (axis > rank) {
+		if (axis >= rank) {
```

Comment:
I see a few lines above in the other split there is apparently the same bug, please fix it there too. Also have a quick grep/find in files of the code to see if this copy-paste error exists elsewhere.